### PR TITLE
[Bug Fix] Edge case: BitSet when highest bit index is a multiplication of 32

### PR DESCRIPTION
**Pull request by yanivshaked**
Hi,

The following test will fail on existing code:
```
    test('bit sets (edge case)', () {
      final counter = BitCounter(0);
      counter.addBitSet(ListSet.fromSorted([0, 2, 5, 2048]));
      expect(counter[2048], 1);
    });
```

The reason for that is that 2048 / 32 is an integer number, so the `hasExtra` value in `_bufferLength32` will be 0.
In this edge case (whenever the highest bit index for bit set is a multiplication of 32), the bit array size will not be sufficient (one bit missing).
The corret fix is to use last bit index +1 for the length of the bit array (see commit).

### CODE REVIEW

File: lib/src/bit_counter.dart [hunk 1]
```diff
@@ -71,8 +71,9 @@ class BitCounter {
   ///
   /// The add starts at the bit position specified by [shiftLeft].
   void addBitSet(BitSet set, {int shiftLeft = 0}) {
-    if (_length < set.length) {
-      _length = set.length;
+    // From last bit index to bit array length: +1
```
Comment (isoos) [inline lib/src/bit_counter.dart]: I'm wondering if the `BitSet.length`'s definition is wrong, especially because the documentation differs on the range between `BitSet` and `BitArray`. Maybe it would be better to rename/change the field and update what it returns? wdyt?

Comment (yanivshaked) [reply lib/src/bit_counter.dart]: Correct. I think the name `length` should remain, but the value returned should be changed:
```
  int get length => _list.isEmpty ? 0 : _list.last + 1;
```
wdyt?

Comment (isoos) [reply lib/src/bit_counter.dart]: Yes, I think that should be the right fix here.

**Comment by yanivshaked** [comment]
I have updated the fix, please review and merge.

**Review by isoos — APPROVED** [review]
